pkg/resources/cruisecontrol/deployment.go

```diff
@@ -41,7 +41,7 @@ func (r *Reconciler) deployment(podAnnotations map[string]string) runtime.Object
 	volume = append(volume, r.KafkaCluster.Spec.CruiseControlConfig.Volumes...)
 	volumeMount = append(volumeMount, r.KafkaCluster.Spec.CruiseControlConfig.VolumeMounts...)
 
-	if r.KafkaCluster.Spec.ListenersConfig.SSLSecrets != nil && util.IsSSLEnabledForInternalCommunication(r.KafkaCluster.Spec.ListenersConfig.InternalListeners) {
+	if r.KafkaCluster.Spec.IsClientSSLSecretPresent() && util.IsSSLEnabledForInternalCommunication(r.KafkaCluster.Spec.ListenersConfig.InternalListeners) {
```

Comment:
Shouldn't be an error returned  when `util.IsSSLEnabledForInternalCommunication` returns `true` while `r.KafkaCluster.Spec.IsClientSSLSecretPresent()` returns `false`

Reply:
The code executing stops sooner with error if this case happen.